src/lib/_proxied-dot-io/fetch-layout-props.ts

```diff
@@ -0,0 +1,26 @@
+import { ComponentType } from 'react'
+import { Products } from '@hashicorp/platform-product-meta'
+
+interface RivetParams {
+  query: string
+  dependencies: ComponentMaybeWithQuery
+}
+
+type ComponentMaybeWithQuery = ComponentType & { rivetParams?: RivetParams }
+
+export default async function fetchLayoutProps(
```

Comment:
Abstracting this out from `_app` so that we can also use it in `_error`, which doesn't call `App.getInitialProps`.